pkg/util/annotation.go

```diff
@@ -29,3 +47,31 @@ func GetAnnotationValue(annotations map[string]string, annotationKey string) str
 	}
 	return annotations[annotationKey]
 }
+
+func getDeletedAnnotationKeys(desired, observed *unstructured.Unstructured) sets.Set[string] {
+	recordKeys := sets.NewString(strings.Split(observed.GetAnnotations()[workv1alpha2.ManagedAnnotation], ",")...)
+	for key := range desired.GetAnnotations() {
+		recordKeys.Delete(key)
+	}
+	return sets.Set[string](recordKeys)
+}
+
+// RecordManagedAnnotations sets or updates the annotation(resourcetemplate.karmada.io/managed-annotations)
+// to record the annotation keys.
+func RecordManagedAnnotations(object *unstructured.Unstructured) {
+	annotations := object.GetAnnotations()
+	if annotations == nil {
+		annotations = make(map[string]string, 1)
+	}
+	// record annotations.
+	managedKeys := []string{workv1alpha2.ManagedAnnotation, workv1alpha2.ManagedLabels}
```

Comment:
Because `ManagedAnnotation` and `ManagedLabels` are propagated through karmada, they need to be record as well.

Reply:
That's why I merged the two functions together in the [first commit](https://github.com/karmada-io/karmada/pull/3099#discussion_r1099782042), so that it's very clear why the two keys need to be recorded in this way.